swimlane/core/adapters/task.py

```diff
@@ -0,0 +1,55 @@
+from swimlane.core.resolver import SwimlaneResolver
+from swimlane.core.cache import check_cache
+from swimlane.core.resources.task import Task
+from swimlane.utils import one_of_keyword_only
+
+
+class TaskAdapter(SwimlaneResolver):
+    """Handles retreival of Swimlane Task Resources and execution of tasks against records."""
+    
+    @check_cache(Task)
+    @one_of_keyword_only('id', 'name')
+    def get(self, key, value):
+        """Get a single task by id or name"""
+        task_list = self.list()
```

Comment:
Why do we need to pull all tasks if we have id? We have API endpoints for Find by Id

Reply:
Yeah good point, I can make that change.

Reply:
I have changed this per your suggestion.